util/helm/client.go

```diff
@@ -98,6 +119,25 @@ type nativeHelmChart struct {
 	indexCache      indexCache
 	proxy           string
 	noProxy         string
+	customUserAgent string // Custom User-Agent string (optional)
+}
+
+// getUserAgent returns the User-Agent string to use for HTTP requests.
+// Priority: code-level (WithUserAgent) > environment variable > default
+func (c *nativeHelmChart) getUserAgent() string {
+	// 1. Code-level custom User-Agent (WithUserAgent option, primarily for testing)
+	if c.customUserAgent != "" {
+		return c.customUserAgent
+	}
+
+	// 2. Global User-Agent from environment variable
+	if envUA := os.Getenv("ARGOCD_HELM_USER_AGENT"); envUA != "" {
+		return envUA
+	}
+
+	// 3. Default User-Agent with version and platform info
+	version := common.GetVersion()
+	return fmt.Sprintf("argocd-repo-server/%s (%s)", version.Version, version.Platform)
 }
```

Comment:
The user agent header should be set from `cmd/argocd-repo-server/commands/argocd_repo_server.go` (see how the other env vars are set there), and the variable propagated into the helm client via `reposerver.NewServer`.